packages/integration-sdk-cli/src/neo4j/neo4jGraphStore.ts

```diff
@@ -0,0 +1,161 @@
+  password: string;
+}
+
+export class Neo4jGraphStore {
+  private neo4jDriver: neo4j.Driver;
+  private persistedSession: neo4j.Session;
+  private databaseName = 'neo4j';
+  private typeList = new Set<string>();
+  private integrationInstanceID: string;
+
+  constructor(instanceID: string, params: Neo4jGraphObjectStoreParams, session?: neo4j.Session) {
+    if(session) {
+      this.persistedSession = session;
+    }
+    else {
+      this.neo4jDriver = neo4j.driver(
+        params.uri,
+        neo4j.auth.basic(params.username, params.password),
+      );
+    }
+    this.integrationInstanceID = instanceID;
+  }
+
+  private async runCypherCommand(cypherCommand: string): Promise<neo4j.Result> {
+    if(this.persistedSession) {
+      const result = await this.persistedSession.run(cypherCommand);
+      return result;
+    }
+    else {
+      const session = this.neo4jDriver.session({
+        database: this.databaseName,
+        defaultAccessMode: neo4j.session.WRITE,
+      });
+      const result = await session.run(cypherCommand);
+      await session.close();
+      return result;
+    }
+  }
+
+  private startsWithNumeric(str: string){
+    return /^\d+$/.test(str);
+  }
+
+  private sanitizePropertyName(propertyName: string): string {
+    let sanitizedName = '';
+    if(this.startsWithNumeric(propertyName)) {
+      sanitizedName += 'n';
+    }
+    sanitizedName += propertyName;
+    sanitizedName = sanitizedName.replace(/[\s!@#$%^&*()-=+\\|'";:/?.,><`~\t\n[\]{}]/g, "_");
+    return sanitizedName;
+  }
+
+  private sanitizeValue(value: string): string {
+    return value.replace(/'/gi, "\\'")
+  }
+
+  public buildPropertyString(propList, nodeName) {
+    let propString = '';
+    for (const key in propList) {
+      if (key === '_rawData') {
+        //stringify JSON in rawData so we can store it.
+        propString += `${nodeName}.${key} = '${JSON.stringify(propList[key])}', `;
+      } else {
+        // Escape single quotes so they don't terminate strings prematurely
+        const finalValue = this.sanitizeValue(propList[key].toString());
+        // Sanitize out characters that aren't allowed in property names
+        const propertyName = this.sanitizePropertyName(key);
+        propString += `${nodeName}.${propertyName} = '${finalValue}', `;
+      }
+    }
+    propString = propString.slice(0, -2);
+
+    return propString;
+  }
+
+  async addEntities(newEntities: Entity[]) {
+    for (const entity of newEntities) {
+      //Add index if not already in types.  This will optimize future
+      //MATCH/MERGE calls.
+      if (!this.typeList.has(entity._type)) {
+        await this.runCypherCommand(
+          `CREATE INDEX index_${entity._type} IF NOT EXISTS FOR (n:${entity._type}) ON (n._key, n._integrationInstanceID);`,
+        );
+        this.typeList.add(entity._type);
+      }
+      const propString = this.buildPropertyString(entity, 'n');
+      const finalKeyValue = this.sanitizeValue(entity._key.toString());
```

Comment:
Should this be calling `sanitizeValue` or `sanitizePropertyName`? It looks like `sanitizeValue` just escapes single-quote characters.

Reply:
I believe it should be sanitizeValue.  All _key values are being stored as strings (so that we can do MATCH/MERGE statements across all nodes and not worry about one random node wanting to set its key to a number or something) and these strings are all surrounded by single quotes.  So the only thing we need to do to sanitize it is ensure that any single quotes are escaped and don't terminate the string prematurely.